File: ios/Runner/Analytics/AnalyticsMethodChannel.swift
import Flutter
import FirebaseAnalytics

class AnalyticsMethodChannel: NSObject, FlutterPlugin {
    private static let channelName = "com.serasa.pokedex/analytics"

    static func register(with registrar: FlutterPluginRegistrar) {
        let channel = FlutterMethodChannel(
            name: channelName,
            binaryMessenger: registrar.messenger()
        )
        let instance = AnalyticsMethodChannel()
        registrar.addMethodCallDelegate(instance, channel: channel)
    }

    func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
        switch call.method {
        case "logEvent":
            handleLogEvent(call, result: result)
        case "logScreenView":
            handleLogScreenView(call, result: result)
        default:
            result(FlutterMethodNotImplemented)
        }
    }

    private func handleLogEvent(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
        guard let args = call.arguments as? [String: Any],
              let eventName = args["eventName"] as? String,
              let parameters = args["parameters"] as? [String: Any] else {
            result(FlutterError(
                code: "INVALID_ARGUMENT",
                message: "Missing parameters",
                details: nil
            ))
            return
        }

        Analytics.logEvent(eventName, parameters: parameters)
        result(nil)
    }

    private func handleLogScreenView(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
        guard let args = call.arguments as? [String: Any],
              let screenName = args["screenName"] as? String,
              let screenClass = args["screenClass"] as? String else {
            result(FlutterError(
                code: "INVALID_ARGUMENT",
                message: "Missing parameters",
                details: nil
            ))
            return
        }

        Analytics.logEvent(
            AnalyticsEventScreenView,
            parameters: [
                AnalyticsParameterScreenName: screenName,
                AnalyticsParameterScreenClass: screenClass
            ]
        )
        result(nil)
    }
}

File: android/app/src/main/kotlin/com/serasa/pokedex/analytics/AnalyticsMethodChannel.kt
package com.serasa.pokedex.analytics

import android.os.Bundle
import com.google.firebase.analytics.FirebaseAnalytics
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel

class AnalyticsMethodChannel(
    private val firebaseAnalytics: FirebaseAnalytics
) : MethodChannel.MethodCallHandler {

    companion object {
        private const val CHANNEL_NAME = "com.serasa.pokedex/analytics"
    }

    override fun onMethodCall(call: MethodCall, result: MethodChannel.Result) {
        when (call.method) {
            "logEvent" -> {
                val eventName = call.argument<String>("eventName")
                val parameters = call.argument<Map<String, Any>>("parameters")

                if (eventName != null && parameters != null) {
                    logEvent(eventName, parameters)
                    result.success(null)
                } else {
                    result.error("INVALID_ARGUMENT", "Missing parameters", null)
                }
            }

            "logScreenView" -> {
                val screenName = call.argument<String>("screenName")
                val screenClass = call.argument<String>("screenClass")

                if (screenName != null && screenClass != null) {
                    logScreenView(screenName, screenClass)
                    result.success(null)
                } else {
                    result.error("INVALID_ARGUMENT", "Missing parameters", null)
                }
            }

            else -> result.notImplemented()
        }
    }

    private fun logEvent(eventName: String, parameters: Map<String, Any>) {
        val bundle = Bundle().apply {
            parameters.forEach { (key, value) ->
                when (value) {
                    is String -> putString(key, value)
                    is Int -> putInt(key, value)
                    is Long -> putLong(key, value)
                    is Double -> putDouble(key, value)
                    is Boolean -> putBoolean(key, value)
                }
            }
        }
        firebaseAnalytics.logEvent(eventName, bundle)
    }

    private fun logScreenView(screenName: String, screenClass: String) {
        val bundle = Bundle().apply {
            putString(FirebaseAnalytics.Param.SCREEN_NAME, screenName)
            putString(FirebaseAnalytics.Param.SCREEN_CLASS, screenClass)
        }
        firebaseAnalytics.logEvent(FirebaseAnalytics.Event.SCREEN_VIEW, bundle)
    }
}